### PR TITLE
feat: `avm/res/desktop-virtualization/application-group`

### PR DESCRIPTION
**Pull request by moisesjgomez**
## Description

Closes avm/res/desktop-virtualization/application-group #146
[avm.res.desktop-virtualization.application-group](https://github.com/moisesjgomez/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.application-group.yml)

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

### CODE REVIEW

**Comment by segraef** [comment]
Please add a status check badge of your latest successful [run](https://github.com/moisesjgomez/bicep-registry-modules/actions/runs/7578157756) as proof.

File: avm/res/network/private-endpoint/tests/e2e/max/dependencies.bicep [hunk 1]
```diff
@@ -35,23 +35,23 @@ resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-04-01' = {
...
+//     }
+//     tenantId: tenant().tenantId
+//     enablePurgeProtection: null
+//     enabledForTemplateDeployment: true
+//     enabledForDiskEncryption: true
+//     enabledForDeployment: true
+//     enableRbacAuthorization: true
+//     accessPolicies: []
+//   }
+// }
```
Comment (segraef) [inline avm/res/network/private-endpoint/tests/e2e/max/dependencies.bicep:54]: If not required please remove.

**Review by segraef — DISMISSED** [review]

File: .github/workflows/avm.res.desktop-virtualization.application-group.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+name: "avm.res.desktop-virtualization.application-group"
```
Comment (eriqua) [inline .github/workflows/avm.res.desktop-virtualization.application-group.yml:1]: Hi @moisesjgomez, also here, codeowners file should be updated. For guidelines on how to configure teams please refer to https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

**Comment by moisesjgomez** [comment]
[avm.res.desktop-virtualization.application-group](https://github.com/moisesjgomez/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.application-group.yml)

File: .github/workflows/avm.res.desktop-virtualization.application-group.yml (hunk 2 above)
Comment (moisesjgomez) [reply .github/workflows/avm.res.desktop-virtualization.application-group.yml:1]: I've created the teams and added the line in the codeowners file.

**Comment by ChrisSidebotham** [comment]
@moisesjgomez - it appears your latest run is failing

File: avm/res/desktop-virtualization/application-group/tests/e2e/defaults/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-desktopvirtualization.applicationgroup-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
```
Comment (ChrisSidebotham) [inline avm/res/desktop-virtualization/application-group/tests/e2e/defaults/main.test.bicep:11]: Due to a change implemented in #838. Please update all `main.test.bicep` files with the following updated 

```suggestion
param resourceLocation string = deployment().location
```

Please note all subsequent refences to the location parameter will need to be updated as per the following example:
` location: resourceLocation`

Comment (AlexanderSehr) [reply avm/res/desktop-virtualization/application-group/tests/e2e/defaults/main.test.bicep:11]: @ChrisSidebotham - should we maybe wait with this until the CI can support it?

File: avm/res/desktop-virtualization/application-group/modules/application.bicep [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      }
+    }
+  }
+}
+
+resource appGroup 'Microsoft.DesktopVirtualization/applicationGroups@2023-09-05' existing = {
+  name: applicationGroupName
+}
+
+resource application 'Microsoft.DesktopVirtualization/applicationGroups/applications@2023-09-05' = {
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/modules/application.bicep:65]: This should be created as a child-module as opposed to a 'nested' template. The main reason is that we may opt to publish child modules in the future + want to keep it consistent with other modules.

To this end, please move this file into a folder `avm/res/desktop-virtualization/application-group/application/main.bicep` that would also contain a ReadMe.md that the `Set-AVMModule` utility can generate for you.

Also note, that the reference in the parent module would need to be updated to the changed folder path.

File: avm/res/desktop-virtualization/application-group/modules/application.bicep [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/modules/application.bicep:60]:
```suggestion
```

Not deployed as part of child-modules

File: avm/res/desktop-virtualization/application-group/modules/application.bicep [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/modules/application.bicep:59]:
```suggestion
```

Not deployed as part of child-modules

File: avm/res/desktop-virtualization/application-group/tests/e2e/max/dependencies.bicep [hunk 7]
```diff
@@ -0,0 +1,34 @@
+@description('Optional. The location to deploy to.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/tests/e2e/max/dependencies.bicep:6]:
```suggestion
```

Seems redundant for the dependency. Should be kept on point.

File: avm/res/desktop-virtualization/application-group/tests/e2e/max/dependencies.bicep [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+  tags: tags
+}
+
+resource hostPool 'Microsoft.DesktopVirtualization/hostPools@2022-09-09' = {
+  name: hostPoolName
+  location: location
+  tags: tags
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/tests/e2e/max/dependencies.bicep:22]:
```suggestion
```
Seems redundant for the dependency. Should be kept on point.

File: avm/res/desktop-virtualization/application-group/tests/e2e/max/main.test.bicep [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    location: resourceLocation
+    hostPoolName: 'dep-${namePrefix}-hp-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
```
Comment (AlexanderSehr) [inline avm/res/desktop-virtualization/application-group/tests/e2e/max/main.test.bicep:42]:
```suggestion
```

Seems redundant for the dependency. Should be kept on point.

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Hey @moisesjgomez - the pipeline badge seems gone as are your workflow runs. I guess you reset your fork? Anyways - code-wise all looks good, just would like to see a pipeline badge to ensure the latest tests pass and we're good to go :)

**Comment by ChrisSidebotham** [comment]
> Hey @moisesjgomez - the pipeline badge seems gone as are your workflow runs. I guess you reset your fork? Anyways - code-wise all looks good, just would like to see a pipeline badge to ensure the latest tests pass and we're good to go :)

Assuming this is due to the issues with GH Login.... @moisesjgomez please update the badge when you are able to do a pipeline run next!

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**